src/Microsoft.AspNet.Mvc.Core/ControllerActionInvoker.cs

```diff
@@ -125,5 +125,22 @@ internal static IActionResult CreateActionResult([NotNull] Type declaredReturnTy
                 DeclaredType = actualReturnType
             };
         }
+
+        private static Type GetTaskInnerTypeOrNull([NotNull] Type type)
```

Comment:
Can this be refactored/merged with generic-type-matcher? Talk to @dougbu if you have questions